InnerEye/ML/deep_learning_config.py

```diff
@@ -608,6 +611,40 @@ def get_path_to_checkpoint(self, epoch: int, for_mean_teacher_model: bool = Fals
                / self.checkpoint_folder \
                / f"{epoch}{filename}"
 
+    def get_recovery_path_train(self, run_recovery: Optional[RunRecovery],
+                                is_mean_teacher: bool, epoch: int) -> Path:
+        if run_recovery:
+            checkpoint_paths = run_recovery.get_checkpoint_paths(epoch, is_mean_teacher)[0]
+        else:
+            checkpoint_paths = self.get_path_to_checkpoint(epoch, is_mean_teacher)
+        return checkpoint_paths
+
+    def get_recovery_path_test(self, run_recovery: Optional[RunRecovery],
+                               is_mean_teacher: bool, epoch: int) -> List[Path]:
+        if run_recovery:
+            checkpoint_paths = run_recovery.get_checkpoint_paths(epoch, is_mean_teacher)
+            checkpoint_exists = []
+            # Discard any checkpoint paths that do not exist - they will make inference/registration fail.
+            # This can happen when some child runs fail; it may still be worth running inference
+            # or registering the model.
+            for path in checkpoint_paths:
+                if path.is_file():
+                    checkpoint_exists.append(path)
+                else:
+                    logging.warning(f"Could not recover checkpoint path {path}")
+
+            if len(checkpoint_exists) > 0:
+                return checkpoint_paths
+
+        logging.warning(f"Using checkpoints from current run, "
+                        f"could not find any run recovery checkpoints for epoch {epoch}")
+        # We found the checkpoint(s) in the run being recovered. If we didn't, it's probably because the epoch
+        # is from the current run, which has been doing more training, so we look for it there.
+        checkpoint_paths = [self.get_path_to_checkpoint(epoch, is_mean_teacher)]
+        if not checkpoint_paths[0].is_file():
+            raise ValueError(f"Could not find checkpoint at path {checkpoint_paths[0]}")
```

Comment:
Everywhere you use this function I see you are catching the exception, and the non-existent checkpoints are a valid scenario, rather than a catastrophe. I'd then rather make this function return None, adhering to the guideline that exceptions should not be used as part of the normal control flow.